### PR TITLE
feat: Allow server and client to specify their own tsconfig

### PR DESCRIPTION
**Pull request by JakeGinnivan**

### CODE REVIEW

File: lib/config/ts-loader-config.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    plugins: webpack.Plugin[],
+    resolvePlugins: webpack.ResolvePlugin[],
+    options: CreateWebpackConfigOptions,
+    buildTarget: BuildTarget,
+): webpack.RuleSetRule {
+    if (!disableTypeCheck) {
+        plugins.push(new ForkTsCheckerWebpackPlugin({ checkSyntacticErrors: true })) // fork ts checking
+    }
+
+    const configFile =
```
Comment (intelseb) [inline lib/config/ts-loader-config.ts]: should there be a fallback to tsconfig.json?

File: lib/bin/build.ts [hunk 2]
```diff
@@ -9,12 +9,7 @@ import { watchtowerConfigFilename } from '../runtime/config/config'
...
 import { webpackPromise } from '../util/webpack'
-import {
-    getMd5,
-    setupWithBuildInfo,
-    TSCONFIG_VALIDATION_ITEM,
-    validateCache,
-} from './cache-validator'
+import { getMd5, setupWithBuildInfo, validateCache } from './cache-validator'
 
 export function smp(buildConfig: BuildConfig, webpackConfig: webpack.Configuration) {
```
Comment (intelseb) [inline lib/bin/build.ts]: can probably remove this in a future pr

File: lib/config/ts-loader-config.ts (hunk 1 above)
Comment (JakeGinnivan) [reply lib/config/ts-loader-config.ts]: Nope, I don't set the configFile props if they are not specified. Then the loaders will just use the default

Comment (intelseb) [reply lib/config/ts-loader-config.ts]: cool!

**Pull request merged**